File: backend-comic/src/main/java/com/thjvjpxx/backend_comic/model/Level.java
package com.thjvjpxx.backend_comic.model;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.fasterxml.jackson.annotation.JsonInclude;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Entity(name = "levels")
@FieldDefaults(level = AccessLevel.PRIVATE)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Level {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "VARCHAR(36)")
    String id;

    @Column(name = "level_number")
    int levelNumber;

    String name;

    String color;

    @Column(name = "exp_required")
    Integer expRequired;

    @Column(name = "url_gif")
    String urlGif;

    @Column(name = "created_at")
    @CreationTimestamp
    LocalDateTime createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "level_type_id")
    LevelType levelType;
}

File: frontend-comic/src/types/level.ts
export interface LevelTypeResponse {
  id: string;
  name: string;
  createdAt: string;
  updatedAt: string;
}

export interface LevelTypeRequest {
  name: string;
}

export interface LevelResponse {
  id: string;
  levelNumber: number;
  name: string;
  color: string;
  expRequired: number;
  urlGif: string;
  levelType: LevelTypeResponse;
  createdAt: string;
  updatedAt: string;
}

export interface LevelRequest {
  name: string;
  levelNumber: number;
  color: string;
  expRequired: number;
  levelTypeId: string;
  urlGif: string;
}


